lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -442,6 +445,14 @@ def ellipsoid(shapeOfTheEarth, major, minor, radius):
             major *= 1000
             minor *= 1000
         result = icoord_systems.GeogCS(major, minor)
+    if shapeOfTheEarth == 4:
+        # Earth assumed oblate spheroid as defined in IAG-GRS80 model.
+        result = icoord_systems.GeogCS(6378137,
+                                       inverse_flattening=298.257222101)
+    if shapeOfTheEarth == 5:
+        # Earth assumed represented by WGS84 (as used by ICAO since 1998).
+        result = icoord_systems.GeogCS(6378137,
+                                       inverse_flattening=298.257223563)
```

Comment:
It's what [we had before](https://github.com/SciTools/iris/blob/f933382cc4eb64311772c3ba83daafa77b280317/lib/iris/fileformats/grib/__init__.py#L500), but also I did check against the [grib documentation](http://apps.ecmwf.int/codes/grib/format/grib2/ctables/3/2) for 2 and 4 (but not 5, that I just relied on what we had before.)